internal/pkg/keeper/client.go

```diff
@@ -69,23 +69,26 @@ func (k *keeperClient) Register() error {
 		},
 	}
 
-	jsonEncodedData, err := json.Marshal(registrationReq)
-	if err != nil {
-		return fmt.Errorf("failed to encode registration request: %s", err.Error())
-	}
-
 	// check if the service registry exists first
 	resp, err := getRegistryByService(k.config.GetRegistryUrl() + ApiRegistrationByServiceIdRoute + k.serviceKey)
 	if err != nil {
 		return fmt.Errorf("failed to check the %s service registry status: %s", k.serviceKey, err.Error())
 	}
 
 	// call the PUT registry API to update the registry if the service already exists
-	// otherwise, call the POST API to create the registry
-	httpMethod := http.MethodPost
-	if resp.StatusCode == http.StatusOK {
-		httpMethod = http.MethodPut
+	// otherwise, call the POST API to create the registry with the initial health check status is UNKNOWN
+	httpMethod := http.MethodPut
+	if resp.StatusCode == http.StatusNotFound {
+		httpMethod = http.MethodPost
+		registrationReq.Registration.Status = Unknown
```

Comment:
to make sure the initial status is `Unkown` for all kinds of circumstances, move this line into the core-keeper POST API